File: 0200-number-of-islands/0200-number-of-islands.cpp
class Solution {
public:
    // vector<vector<int>> visited;
    int dfs(int i, int j, vector<vector<char>>& grid){
        if(i>=0 && j>=0 && i<grid.size() && j<grid[0].size() && grid[i][j]=='1'){
            grid[i][j]='0';
            dfs(i-1, j, grid);
            dfs(i, j-1, grid);
            dfs(i+1, j, grid);
            dfs(i, j+1, grid);
        }
        return 1;
    }
    int numIslands(vector<vector<char>>& grid) {
        int n= grid.size(), m= grid[0].size(), count=0;
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(grid[i][j]=='1'){
                    dfs(i,j,grid);
                    count++;
                }
            }
        }
        return count;
    }
};

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        count = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]=="1":
                    count += 1
                    def makeIslandSea(m, n):
                        if m>=0 and m<len(grid) and n>=0 and n<len(grid[0]) and grid[m][n]=="1":
                            grid[m][n]="0"
                            print("m :",m ,"  n: ",n)
                            makeIslandSea(m-1, n)
                            makeIslandSea(m, n-1)
                            makeIslandSea(m+1, n)
                            makeIslandSea(m, n+1)
                    makeIslandSea(i,j)
                            
        return count

                            
                 
                            
        